intermediate_source/compiled_autograd_tutorial.rst

```diff
@@ -0,0 +1,301 @@
+      loss.backward()
+
+There should be a ``call_hook`` node in the graph, which dynamo will later inline into
+
+.. code:: python
+
+   stderr_output = """
+   DEBUG:torch._dynamo.compiled_autograd.__compiled_autograd_verbose:Cache miss due to new autograd node: torch::autograd::GraphRoot (NodeCall 0) with key size 39, previous key sizes=[]
+   DEBUG:torch._dynamo.compiled_autograd.__compiled_autograd_verbose:TRACED GRAPH
+   ===== Compiled autograd graph =====
+   <eval_with_key>.2 class CompiledAutograd(torch.nn.Module):
+      def forward(self, inputs, sizes, scalars, hooks):
+         ...
+         getitem_2 = hooks[0];  hooks = None
+         call_hook: "f32[10, 10][0, 0]cpu" = torch__dynamo_external_utils_call_hook(getitem_2, aot0_expand, hook_type = 'tensor_pre_hook');  getitem_2 = aot0_expand = None
+         ...
+   """
+
+Common recompilation reasons for Compiled Autograd
+--------------------------------------------------
+1. Due to change in autograd structure 
+
+.. code:: python
+
+   torch._dynamo.config.compiled_autograd = True
+   x = torch.randn(10, requires_grad=True)
+   for op in [torch.add, torch.sub, torch.mul, torch.div]:
+      loss = op(x, x).sum()
+      torch.compile(lambda: loss.backward(), backend="eager")()
+
+You should see some recompile messages: **Cache miss due to new autograd node**.
+
+.. code:: python
+
+   stderr_output = """
+   Cache miss due to new autograd node: torch::autograd::GraphRoot (NodeCall 0) with key size 39, previous key sizes=[] 
+   ...
+   Cache miss due to new autograd node: SubBackward0 (NodeCall 2) with key size 56, previous key sizes=[]
+   ...
+   Cache miss due to new autograd node: MulBackward0 (NodeCall 2) with key size 71, previous key sizes=[]
+   ...
+   Cache miss due to new autograd node: DivBackward0 (NodeCall 2) with key size 70, previous key sizes=[]
+   ...
+   """
+
+2. Due to dynamic shapes
+
+.. code:: python
+
+   torch._dynamo.config.compiled_autograd = True
+   for i in [10, 100, 10]:
+      x = torch.randn(i, i, requires_grad=True)
+      loss = x.sum()
+      torch.compile(lambda: loss.backward(), backend="eager")()
+
+You should see some recompiles messages: **Cache miss due to changed shapes**.
+
+.. code:: python
+
+   stderr_output = """
+   ...
+   Cache miss due to changed shapes: marking size idx 0 of torch::autograd::GraphRoot (NodeCall 0) as dynamic
+   Cache miss due to changed shapes: marking size idx 1 of torch::autograd::AccumulateGrad (NodeCall 2) as dynamic
+   Cache miss due to changed shapes: marking size idx 2 of torch::autograd::AccumulateGrad (NodeCall 2) as dynamic
+   Cache miss due to changed shapes: marking size idx 3 of torch::autograd::AccumulateGrad (NodeCall 2) as dynamic
+   ...
+   """
+
+Conclusion
+----------
+In this tutorial, we went over the high-level ecosystem of ``torch.compile`` with compiled autograd, the basics of compiled autograd and a few common recompilation reasons.
```

Comment:
Where should user go from here. Can we add something like:

To learn more about Compiled Autograd, go to ....